### ISSUE TITLE
[Ansible] How install different kubernetes version clusters on fedora/rhel

### ISSUE THREAD

**Issue by mdshuai** [kubernetes/contrib#1442]
Sometime I want to install a latest k8s version, like( v1.3.3 ,  v1.4.0-alpha.1), I notice there is a version parameter https://github.com/kubernetes/contrib/blob/master/ansible/roles/kubernetes/defaults/main.yml#L2
But if I change it to different version, it still install a k8s cluster with v1.2.
How to install different version k8s cluster with ansible on fedora/rhel? How to install k8s cluster from github release with ansible on fedora/rhel? thanks so much.

**Comment by ingvagabund** [comment]
> Sometime I want to install a latest k8s version, like( v1.3.3 , v1.4.0-alpha.1), I notice there is a version parameter https://github.com/kubernetes/contrib/blob/master/ansible/roles/kubernetes/defaults/main.yml#L2
> But if I change it to different version, it still install a k8s cluster with v1.2.

Current practice is to install kubernetes from distribution builds (see [1]). Retrieval of kubernetes binaries from github repository is carried out for CoreOS and Ubuntu only [2]. 

> How to install different version k8s cluster with ansible on fedora/rhel?

Currently, you can install only the latest kubernetes rpms.

> How to install k8s cluster from github release with ansible on fedora/rhel?

Currently, you can not.

Though, in both cases it would not be hard to extend the playbook to support both scenarios. Let me take a look at it.

[1] https://github.com/kubernetes/contrib/blob/master/ansible/group_vars/all.yml#L7
[2] https://github.com/kubernetes/contrib/blob/master/ansible/roles/kubernetes/tasks/main.yml#L35

**Comment by ingvagabund** [comment]
This could be fixed as a part of https://github.com/kubernetes/contrib/pull/1444 kubernetes roles refactoring.

**Comment by mdshuai** [comment]
@ingvagabund thanks for your comment

**Comment by ingvagabund** [comment]
**kubernetes upstream release based installation** (e.g. ``kubernetes-1.3.1``):

```
ansible-playbook \ 
  -i INVENTORY playbooks/deploy-cluster.yml \
  --extra-vars "kube_source_type=github-release kube_version=1.3.1"
```

**kubernetes distribution rpm based installation** (e.g. ``kubernetes-1.3.0-0.2.git507d3a7.fc26``):

```
ansible-playbook \ 
  -i INVENTORY playbooks/deploy-cluster.yml \
  --extra-vars "\
  kube_source_type=distribution-rpm \
  kube_rpm_url_base=https://kojipkgs.fedoraproject.org//packages/kubernetes/1.3.0/0.2.git507d3a7.fc26/x86_64 \
  kube_rpm_url_sufix=1.3.0-0.2.git507d3a7.fc26.x86_64.rpm \
  "
```

or basically any build (scratch build) with standard rpm url paths, i.e.:
* ``KUBE_RPM_URL_BASE/kubernetes-client/KUBE_RPM_URL_SUFIX``
* ``KUBE_RPM_URL_BASE/kubernetes-master/KUBE_RPM_URL_SUFIX``
* ``KUBE_RPM_URL_BASE/kubernetes-node/KUBE_RPM_URL_SUFIX``

As long as you much builds of your distribution (e.g. Fedora, RHEL, CentOS), it gets installed.

**Comment by ingvagabund** [comment]
@mdshuai implemented via #1444 [1]. If it is still not enough, please open another issue.

[1] https://github.com/kubernetes/contrib/tree/master/ansible#kubernetes-source-type

**Comment by mdshuai** [comment]
@ingvagabund thanks.
